src/components/Header/Header.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  FC,
+  useState,
+  useCallback,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import { createTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onAddTodo: Dispatch<SetStateAction<Todo[]>>;
+  setError: (error: boolean) => void;
+};
+
+export const Header: FC<Props> = ({ onAddTodo, setError }) => {
+  const [title, setTitle] = useState('');
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+  const user = JSON.parse(localStorage.getItem('user')!) || {
+    id: 0,
+    name: '',
+    username: '',
+    email: '',
+    phone: '',
+  };
+
+  const handleChangeEvent = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setTitle(value);
+  };
+
+  const handleSubmit = useCallback(
+    async (e: React.FormEvent<HTMLFormElement>) => {
+      try {
+        e.preventDefault();
+
+        if (!title.trim().length) {
```

Comment:
No need to check on length, it should be enough:
```suggestion
        if (!title.trim()) {
```